calculator/src/main/java/com/market/banica/calculator/exception/GlobalExceptionHandler.java

```diff
@@ -20,17 +24,20 @@
  */
 
 @ControllerAdvice
-public class GlobalExceptionHandler {
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(GlobalExceptionHandler.class);
 
-    @ExceptionHandler({FeatureNotSupportedException.class})
-    public ResponseEntity<Object> notSupportedFeatureHandling(FeatureNotSupportedException exception) {
-        return new ResponseEntity<>(Collections.singletonMap("error", exception.getMessage()), HttpStatus.valueOf(501));
+    @ExceptionHandler(IllegalArgumentException.class)
+    protected ResponseEntity<Object> handleIllegalArgument(
+            RuntimeException ex, WebRequest request) {
```

Comment:
it is always better to name the variables to be more self-explainable - for example exception or runtimeException